components/managedidentity/locals.tf

```diff
@@ -1,5 +1,18 @@
 locals {
   mi_environment = var.env == "ptlsbox" ? "cftsbox-intsvc" : var.env == "ptl" ? "cftptl-intsvc" : var.env == "sbox" ? "sandbox" : var.env == "stg" ? "aat" : var.env == "dev" ? "preview" : var.env == "test" ? "perftest" : var.env
+  cosmosdb_name  = contains(["sandbox", "aat", "preview", "perftest", "dev", "sbox"], var.env) ? "sandbox-pipeline-metrics" : "pipeline-metrics"
+  cosmosdb_rg    = contains(["sandbox", "aat", "preview", "perftest", "dev", "sbox"], var.env) ? "pipelinemetrics-database-sandbox" : "pipelinemetrics-database-prod"
+  cosmosdb_env   = contains(["sbox", "dev"], var.env) ? "sandbox" : "prod"
```

Comment:
Really close now. Think this just needs clearing up, so far cosmosdb_name and cosmosdb_rg suggest that if the env is sbox,aat,preview,perftest, or dev, they will be getting access to the sandbox-pipeline-metrics db

But cosmosdb_env suggests that only sbox and dev should be talking to this db.